metrics/internal/cache/rbd-mirror.go

```diff
@@ -147,10 +148,23 @@ func (s *RBDMirrorStore) WithRBDCommandInput(namespace string) error {
 	return nil
 }
 
-func initCeph(kubeclient clientset.Interface, namespace string) (cephMonitorConfig, error) {
-	secret, err := kubeclient.CoreV1().Secrets(namespace).Get(context.TODO(), "rook-ceph-mon", v1.GetOptions{})
-	if err != nil {
-		return cephMonitorConfig{}, fmt.Errorf("failed to get secret in namespace %q: %v", namespace, err)
+func initCeph(kubeclient clientset.Interface, allowedNamespaces []string) (cephMonitorConfig, error) {
```

Comment:
This function does not need to change. It's job is to fetch the config from a given namespace.
What we should change is, how it's called in different places.

The idea is to store a map of namespace to cephMonitorConfig per store and reuse it. This map should include namespaces in the allowedNamespaces list only. If a namespace is not in the list or we can't find the config in that namespace, we ignore it.